src/main/java/mate.academy/lib/Injector.java

```
@@ -1,13 +1,76 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (interfaceClazz.getClass().isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Missing annotation @Component form class: "
+                    + interfaceClazz.getName());
+        }
```

This check for the `@Component` annotation is not working correctly. The expression `interfaceClazz.getClass()` returns `java.lang.Class.class`, not the class you intend to check. Furthermore, the logic is inverted; it throws an exception if the annotation *is* present, while the message says it's missing.

This check should be performed on the implementation class (`clazz`) after it's found, and it should throw an exception if the annotation is *not* present. This violates checklist item #4.